osutil/disks/disks_linux.go

```diff
@@ -632,6 +649,54 @@ func (d *disk) populatePartitions() error {
 			}
 			part.KernelDeviceNode = devname
 
+			// we should always have the partition type
+			partType, ok := udevProps["ID_PART_ENTRY_TYPE"]
+			if !ok {
+				return fmt.Errorf("cannot get udev properties for device %s (a partition of %s), missing required udev property \"ID_PART_ENTRY_TYPE\"", partDev, d.Dev())
+			}
+
+			// on dos disks, the type is formatted like "0xc", when we prefer to
+			// always use "0C" so fix the formatting
+			if d.schema == "dos" {
+				partType = strings.TrimPrefix(partType, "0x")
```

Comment:
I guess it's safer to make `partType` lowercase, or remove `0X` too here.

Reply:
done